src/averageAges.js

```diff
@@ -55,10 +61,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter(woman =>
+    people.some(mother => (woman.name === mother.mother && woman.sex === 'f')));
+
+  const children = people.filter(child =>
```

Comment:
The condition in the filter function should check if the person's mother matches any of the mothers in the 'mothers' array, not if the person's mother matches any mother.